### PR TITLE
chore: upgrade to Tauri 2.0 Beta

### LINKED ISSUE #108: Upgrade to Tauri 2.0 Beta
### Description

Tauri **v2.0.0-beta** has been released. See [this blog post](https://beta.tauri.app/blog/tauri-2-0-0-beta/).

### Motivation

We should upgrade for stability, new features and preparation for stable release.

### Requirements

1. read requirements & release notes
2. migrate APIs, etc.
3. test builds for Android & iOS

### Open Questions

_No response_

### Are you planning to contribute this in a PR?

Yes

### PR DESCRIPTION
**Pull request by daniel-mader**
# Description of change
Migrate from Tauri 2.0 alpha to beta.

Additionally, a few minor changes have been made:
- update README, add banner image
- remove editor-specific plugin suggestions for VS Code
- remove `Cargo.lock` from unime folder (using workspace lockfile instead)
- add two icons to version control to make the setup step "generate icons" optional (app doesn't start without icons)

## Links to any relevant issues
#108 

## How the change has been tested
Locally running the app on desktop.

## Definition of Done checklist
Add an `x` to the boxes that are relevant to your changes.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by daniel-mader** [comment]
Currently waiting for [this issue](https://github.com/tauri-apps/tauri/pull/8777) being released in the next version (possibly `tauri 2.0.0-beta.3`)

**Comment by daniel-mader** [comment]
> Currently waiting for [this issue](https://github.com/tauri-apps/tauri/pull/8777) being released in the next version (possibly `tauri 2.0.0-beta.3`)

Got released, so we're clear to update.

**Review by nanderstabel — DISMISSED** [review]

File: README.md [hunk 1]
```diff
@@ -25,121 +45,27 @@ npm run tauri ios init
 npm run tauri ios dev
 ```
 
-Before commiting, please make sure to run
+## Contributing
+
+> [!NOTE]
+> Before committing, please make sure to run:
```
Comment (Oran-Dan) [inline README.md]: add backend tests and formatting as well

**Review by Oran-Dan — CHANGES_REQUESTED** [review]
just a few lines in the README and it should be good to go

**Review by Oran-Dan — DISMISSED** [review]

**Review by Oran-Dan — DISMISSED** [review]

**Review by Oran-Dan — APPROVED** [review]

**Review by nanderstabel — APPROVED** [review]

**Pull request merged**